### PR TITLE
Automatically set the executable flag

### PR DESCRIPTION
**Pull request by Cito**
Also some refactoring was necessary.

### CODE REVIEW

File: scripts/update_template_files.py [hunk 1]
```diff
@@ -177,28 +219,30 @@ def main(check: bool = False):
...
-        "Successfully validated the template files."
-        if check
-        else "Successfully updated the template files."
-    )
+    if remove_files(files_to_remove, check=check):
+        updated = True
+
+    if check:
+        if updated:
+            echo_failure("Validating the files from the template failed.")
```
Comment (mephenor) [inline scripts/update_template_files.py:237]:
```suggestion
            echo_failure("Validating files from template failed.")
```

File: scripts/update_template_files.py [hunk 2]
```diff
@@ -177,28 +219,30 @@ def main(check: bool = False):
...
+            sys.exit(1)
+        echo_success("Successfully validated the files from the template.")
```
Comment (mephenor) [inline scripts/update_template_files.py:239]:
```suggestion
        echo_success("Successfully validated files from template.")
```

File: scripts/update_template_files.py [hunk 3]
```diff
@@ -177,28 +219,30 @@ def main(check: bool = False):
...
+    else:
+        echo_success(
+            "The files from the template have been updated."
```
Comment (mephenor) [inline scripts/update_template_files.py:242]:
```suggestion
            "Successfully updated files from template."
```

**Review by mephenor — APPROVED** [review]

**Comment by Cito** [comment]
Also added a small fix

**Review by mephenor — DISMISSED** [review]

**Review by KerstenBreuer — APPROVED** [review]

**Pull request merged**